tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -780,6 +759,147 @@ private ApplicationRpcServer setupRPCService(String hostname) {
     return rpcServer;
   }
 
+  private class TaskScheduler {
```

Comment:
Move this to a new file? AM class is large enough, it would also be easier to unit test TaskScheduler if it is in another class (w/ proper constructors for testing)

Reply:
we need some unit testing for this  class as well..

Reply:
definitely. Also add documentation on why we have capabilities here that are duplicating Azkaban's DAG management capabilities

Reply:
we are not duplicating Azkaban, let's not mention that in documentation... have some mini DAG engine in any framework is expected.